tests/phpunit/tests/functions.php

```diff
@@ -1096,6 +1106,21 @@ public function test_wp_extract_urls() {
 		$this->assertSame( array_slice( $original_urls, 0, 8 ), $urls );
 	}
 
+	/**
+	 * @ticket 30580
+	 *
+	 * Tests for backward compatibility of `wp_extract_urls` to remove unused semicolons.
+	 */
+	public function test_wp_extract_urls_remove_semicolon() {
+		$url_with_typo = 'http://typo.com;';
+		$expected      = array(
+			'http://typo.com',
+		);
+
+		$urls = wp_extract_urls( $url_with_typo );
+		$this->assertSame( $urls, $expected );
```

Comment:
The order of arguments to `assertSame()` is `$expected`, then `$actual`.

Currently, if this assertion failed, the result would be like this:
```
--- Expected
+++ Actual

- Something from $urls
+ Something from $expected
```
As this confuses things when debugging, I suggest swapping the order of arguments to `$expected, $urls`.

---

This isn't necessary, but you can also reduce this test method to:
```php
public function test_wp_extract_urls_remove_semicolon() {
	$expected = array( 'http://typo.com' );
	$actual   = wp_extract_urls( 'http://typo.com;' );

	$this->assertSame( $expected, $actual );
}
```

Reply:
Good catch, fixed!